### PR TITLE
Add CLA

### PR DESCRIPTION
**Pull request by dheater**
Add a Contributor License Agreement to the project.

Address #1043 

I generated a Contributor License Agreement using the [Harmony](http://www.harmonyagreements.org/about.html) CLA generator.

Once approved, I recommend configuring [CLA Assistant](https://cla-assistant.io/) to handle authenticating, tracking signatories, and automating CLA acceptance enforcement through a GitHub Pull Request bot. [CLA Assistant](https://cla-assistant.io/) is a free service.

### CODE REVIEW

File: docs/CLA.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+2.3 Outbound License
+
+As a condition on the grant of rights in Sections 2.1 and 2.2, We agree to license the Contribution only under the terms of the license or licenses which We are using on the Submission Date for the Material (including any rights to adopt any future version of a license if permitted).
+
+In addition, We may use the following licenses for Media in the Contribution: Creative Commons Attribution 4.0 International License (including any right to adopt any future version of a license if permitted).
```
Comment (dheater) [inline docs/CLA.md:41]: I added this reference to the CC Attribution License to cover media files that are not covered by the Apache 2.0 software license. Feel free to recommend alternative media licenses.

File: docs/CLA.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+## 3. Agreement
+
+You confirm that:
+
+(a) You have the legal authority to enter into this Agreement.
+
+(b) You own the Copyright and patent claims covering the Contribution which are required to grant the rights under Section 2.
+
+(c) The grant of rights under Section 2 does not violate any grant of rights which You have made to third parties, including Your employer. If You are an employee, You have had Your employer approve this Agreement or sign the Entity version of this document. If You are less than eighteen years old, please have Your parents or guardian sign the Agreement.
```
Comment (asmodai27) [inline docs/CLA.md:57]: The need to have your employer sign an agreement depends on the contract you have with them (e.g whether it has a specific clause preventing you to do other work, or a clause granting them copyright and/or patenting rights on your work). I'd keep the first sentence but add nuance to the second one.

File: docs/CLA.md (hunk 1 above)
Comment (martukas) [reply docs/CLA.md:41]: We have third-party copyrighted CAD models in the manufacturing directories. We were given informal permission by McMaster-Carr (by email correspondence) to use those, but they didn't want special mention or anything. Any impact here?

**Review by martukas — COMMENTED** [review]
Will there be a way to use CLA assistant to go back and get signatures from our past contributors? I believe we have a pretty decent accounting of who those are and how to reach them.

File: docs/CLA.md (hunk 1 above)
Comment (dheater) [reply docs/CLA.md:41]: IANAL, but my understanding is that they retain copyright (true for all contributors). The essence of the CLA is that contributors are giving us the right to use and distribute the contributed works.  The question, I think, would be do we only have the right to "use", or do we have the right to "use and distribute (per clause 2.1- b)"?

**Comment by martukas** [comment]
Actually, it may not be a bad idea to have it somewhere in the repo? Like, I imagine, once we do adopt this, we should add some words about it to our wiki and have it easily findable/discoverable on demand.

File: docs/CLA.md (hunk 2 above)
Comment (dheater) [reply docs/CLA.md]: added a qualifying phrase to the second sentence.

> If You are an employee whose employment agreement asserts intellectual property rights over your contributions,

File: docs/CLA.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+Thank you for your interest in contributing to RespiraWorks ("We" or "Us").
+
+This contributor agreement ("Agreement") documents the rights granted by contributors to Us. To make this document effective, please sign it and send it to Us by electronic submission, following the instructions at https://cla-assistant.io/. This is a legally binding document, so please read it carefully before agreeing to it. The Agreement may cover more than one software project managed by Us.
+
+## 1. Definitions
+
+"You" means the individual who Submits a Contribution to Us.
+
+"Contribution" means any work of authorship that is Submitted by You to Us in which You own or assert ownership of the Copyright. If You do not own the Copyright in the entire work of authorship, please follow the instructions in https://cla-assistant.io/.
```
Comment (dheater) [inline docs/CLA.md:11]: The link to https://cla-assistant.io/ is a placeholder. This should be updated to reference an page on our site that provides instructions.

File: docs/CLA.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+You confirm that:
+
+(a) You have the legal authority to enter into this Agreement.
+
+(b) You own the Copyright and patent claims covering the Contribution which are required to grant the rights under Section 2.
+
+(c) The grant of rights under Section 2 does not violate any grant of rights which You have made to third parties, including Your employer. If You are an employee, You have had Your employer approve this Agreement or sign the Entity version of this document. If You are less than eighteen years old, please have Your parents or guardian sign the Agreement.
+
+(d) You have followed the instructions in https://cla-assistant.io/, if You do not own the Copyright in the entire work of authorship Submitted.
```
Comment (dheater) [inline docs/CLA.md:59]: The link to https://cla-assistant.io/ is a placeholder. This should be updated to reference an page on our site that provides instructions.

**Review by dheater — COMMENTED** [review]
> Will there be a way to use CLA assistant to go back and get signatures from our past contributors? I believe we have a pretty decent accounting of who those are and how to reach them.

I'm not sure. Looking at the CLA assistant report there is a FAQ that asks about import/export, but really only answers the export question: https://github.com/cla-assistant/cla-assistant#where-can-i-see-the-list-of-signees-is-there-a-way-to-importexport-the-signee-data

The later question about how to add a bot hints that it is possible to add a Github username via the dashboard: https://github.com/cla-assistant/cla-assistant#where-can-i-see-the-list-of-signees-is-there-a-way-to-importexport-the-signee-data

Worst case, I think, is that contributors would have to click to accept the next time they submit a PR.

**Comment by dheater** [comment]
> Actually, it may not be a bad idea to have it somewhere in the repo? Like, I imagine, once we do adopt this, we should add some words about it to our wiki and have it easily findable/discoverable on demand.

Agree. I'll provide an updated PR that adds the CLA to the documentation and provides an instructions page that I mention in my comments about https://cla-assistant.io/ being a placeholder link. I'll update the link at that time too.

File: docs/CLA.md (hunk 1 above)
Comment (martukas) [reply docs/CLA.md:41]: They saw our repo and they said what we are doing was ok.

**Comment by dheater** [comment]
> Actually, it may not be a bad idea to have it somewhere in the repo? Like, I imagine, once we do adopt this, we should add some words about it to our wiki and have it easily findable/discoverable on demand.

Added to the repo along with a "Contributing" page in the documentation that explains a little about the CLA and links to the text.

File: docs/CLA.md (hunk 3 above)
Comment (dheater) [reply docs/CLA.md]: Fixed to reference the new "Contributing" page.

File: docs/CLA.md (hunk 4 above)
Comment (dheater) [reply docs/CLA.md]: Fixed to reference the new "Contributing" page.

**Review by inceptionev — APPROVED** [review]
This looks reasonable to me, and there aren't any issues that stand out.

**Pull request merged**